### PR TITLE
Fix catmatvec

### LINKED ISSUE #2526: AFNI CatMatvec: Caching fails and output file undefined
### Summary
AFNI CatMatvec  interface caching fails and output file is undefined
### How to replicate the behavior
```Python
from nipype.interfaces import afni
from nipype.caching import Memory
memory = Memory('/tmp/test_catmatvec/')
catmatvec = memory.cache(afni.CatMatvec)
out_catmatvec = catmatvec(in_file=[('/tmp/test_caching/allineate_affine.aff12.1D', 'ONELINE')], out_file='/tmp/test_catmatvec2.aff12.1D')
print(out_catmatvec.outputs)
```
gives `out_file = <undefined>`. Moreover, when launching the same code twice the computations are restarted
### Platform details:

{'nibabel_version': '2.2.1', 'sys_executable': '/home/salma/anaconda2/bin/python', 'networkx_version': '2.0', 'numpy_version': '1.13.3', 'sys_platform': 'linux2', 'sys_version': '2.7.14 |Anaconda custom (64-bit)| (default, Oct 16 2017, 17:29:19) \n[GCC 7.2.0]', 'commit_source': 'repository', 'commit_hash': 'e446466', 'pkg_path': '/home/salma/CODE/nipype/nipype', 'nipype_version': '1.0.3-dev+ge446466', 'traits_version': '4.6.0', 'scipy_version': '1.0.0'}
1.0.3-dev+ge446466

### PR DESCRIPTION
**Pull request by salma1601**
Fixes #2526  .

Changes proposed in this pull request
- use `name_template` and `name_source` for generating the out_file
- add `_list_outputs` method

### CODE REVIEW

**Comment by mgxd** [comment]
@salma1601 have you tried this without adding the `_list_outputs` method? I believe it should still work

**Comment by salma1601** [comment]
>@salma1601 have you tried this without adding the _list_outputs method? I believe it should still work

yes and you are right, it works without it ! removed

File: nipype/interfaces/afni/utils.py [hunk 1]
```diff
@@ -616,7 +619,6 @@ def _format_arg(self, name, spec, value):
                                             for i in value]))
         return super(CatMatvec, self)._format_arg(name, spec, value)
 
-
```
Comment (effigies) [inline nipype/interfaces/afni/utils.py]: Could you re-add this line?

Comment (salma1601) [reply nipype/interfaces/afni/utils.py]: yes sorry

**Pull request merged**